data-processing/scripts/infomedia/convert_infomedia_to_jsonlgz.py

```diff
@@ -0,0 +1,134 @@
+'''
+Converting infomedia dataset.
+
+ndjson -> json.gz:
+
+{
+    "id": "...",             # MANDATORY: source-specific identifier
+    "text": "foo",           # MANDATORY: textual content of the document
+    "source": "...",         # MANDATORY: source of the data, such as peS2o, common-crawl, etc.
+    "added": "...",          # OPTIONAL: timestamp we acquired this data (time file was created), specified as YYYY-MM-DDTHH:MM:SS.TIMEZONE e.g 2021-04-13T12:52:46.000Z
+    "created": "..."         # OPTIONAL: timestamp when orig document was created (best-guess if not available), should be specified as a range; "YYYY-MM-DDTHH:MM:SS.TIMEZONE, YYYY-MM-DDTHH:MM:SS.TIMEZONE"
+    "metadata": {            # OPTIONAL: source-specific metadata
+										"sub-source": "...", # OPTIONAL: E.g. "newspaper_ocr"
+										...
+								}        
+}
+'''
+
+import os
+import re
+import json
+import gzip
+import uuid
+import tempfile
+from datetime import datetime, timedelta
+from tqdm import tqdm
+from multiprocessing import Pool, cpu_count
+
+def format_created_range(publish_date):
+    """Create a formatted string representing a time range starting from `publish_date`."""
+    start_date = datetime.strptime(publish_date, "%Y-%m-%dT%H:%M:%SZ")
+    end_date = start_date + timedelta(days=365)
+    return f"{start_date.strftime('%Y-%m-%dT%H:%M:%SZ')}, {end_date.strftime('%Y-%m-%dT%H:%M:%SZ')}"
+
+def remove_html_tags(text):
+    """Remove HTML tags from a string."""
+    html_tag_pattern = re.compile('<.*?>')
+    clean_text = re.sub(html_tag_pattern, '', text)
+    return clean_text
```

Comment:
@jankounchained how did you do the html filtering when you did it?

Reply:
@KennethEnevoldsen @TTTTao725 sorry missed this!
the regex pattern looks good.

if you want to be extra careful, then you control for two edge cases:
- add a multiline flag in case there is a long html tag between mulitple lines
- replace html tags with a single shitespace in case that this happens: `word<br /></p><p id=\"p1\">word`
- take care of excess whitespace with another function

so something like this:
```
def remove_html_tags(text: str) -> str:
    """Remove HTML tags from a string."""
    html_tag_pattern = re.compile('<.*?>', flags=re.MULTILINE)
    clean_text = re.sub(html_tag_pattern, " ", text)
    return clean_text

def remove_whitespace(text: str) -> str:
    """remove excess whitespace from text fields"""
    pat_ws = re.compile(pattern=r"\s\s+", flags=re.MULTILINE)
    clean_text = re.sub(html_tag_pattern, " ", text)
    return clean_text
```